src/Process/Pool/Logger/Message/SerializableProcess.php

```diff
@@ -116,8 +122,36 @@ public function setTypeCode(string $type_code) : SerializableProcessInterface
         return $this;
     }
 
+    public function getMemoryUsageBytes() : int
+    {
+        $this->memory_usage_bytes = memory_get_usage();
+
+        return $this->memory_usage_bytes;
+    }
+
+    public function getMemoryPeakUsageBytes() : int
+    {
+        $this->memory_peak_usage_bytes = memory_get_peak_usage();
+
+        return $this->memory_peak_usage_bytes;
+    }
+
+    public function getMemoryLimitBytes() : int
+    {
+        $this->memory_limit_bytes = $mem_limit = $this->dataUnitToBytes(ini_get('memory_limit'));
+        return $this->memory_limit_bytes;
+    }
+
     public function jsonSerialize()
     {
-      return get_object_vars($this);
+        return get_object_vars($this);
+    }
+
+    /* converts a number with byte unit (B / K / M / G) into an integer */
+    protected function dataUnitToBytes($s)
+    {
+        return (int)preg_replace_callback('/(\-?\d+)(.?)/', function ($m) {
```

Comment:
Is the `\-?` part of the regex for `-1`? Will that interfere with other things? (i.e. something checking if `$x < $serializableProcess->getMemoryLimitBytes()`)